tests/lib/openstack_query/query_blocks/test_query_builder.py

```diff
@@ -10,72 +10,107 @@
 from tests.lib.openstack_query.mocks.mocked_props import MockProperties
 
 
-@pytest.fixture(name="mock_client_side_handlers")
-def client_side_handlers_fixture():
+@pytest.fixture(name="mock_invalid_client_handler")
+def mock_invalid_client_handler_fixture():
     """
-    Returns a dictionary of mocked client_side_handler objects
+    Return a client handler - mocks one handler which does not contain preset
     """
-    return {"mock_client_handler_1": MagicMock(), "mock_client_handler_2": MagicMock()}
+    mock_invalid_client_handler = MagicMock()
+    mock_invalid_client_handler.preset_known.return_value = False
+    mock_invalid_client_handler.check_supported.return_value = False
+    return mock_invalid_client_handler
+
+
+@pytest.fixture(name="mock_valid_client_handler")
+def mock_valid_client_handler_fixture():
+    """
+    Return a client handler - mocks one handler which does contain preset
+    """
+
+    def _mock_preset_known(preset):
+        """A dummy function to mock the preset_known method on the client handler"""
+        return preset in [MockQueryPresets.ITEM_1, MockQueryPresets.ITEM_3]
+
+    def _mock_check_supported(preset, prop):
+        """A dummy function to mock the check_supported method on the client handler"""
+        return preset == MockQueryPresets.ITEM_1 and prop == MockProperties.PROP_1
+
+    mock_invalid_client_handler = MagicMock()
+    mock_invalid_client_handler.preset_known = MagicMock(wraps=_mock_preset_known)
+    mock_invalid_client_handler.check_supported = MagicMock(wraps=_mock_check_supported)
+    return mock_invalid_client_handler
 
 
 @pytest.fixture(name="mock_server_side_handler")
-def server_side_handler_fixture():
+def mock_valid_server_handler_fixture():
     """
-    Returns a mocked server_side_handler_object
+    Return a server-side handler
     """
     return MagicMock()
 
 
 @pytest.fixture(name="instance")
-def instance_fixture(mock_client_side_handlers, mock_server_side_handler):
+def instance_fixture(
+    mock_server_side_handler, mock_invalid_client_handler, mock_valid_client_handler
+):
     """
     Returns an instance with mocked client_side_handlers and
     server_side_handler injects
     """
-    mock_prop_enum_cls = MockProperties
-    instance = QueryBuilder(
-        prop_enum_cls=mock_prop_enum_cls,
-        client_side_handlers=list(mock_client_side_handlers.values()),
+
+    client_handlers = [mock_invalid_client_handler, mock_valid_client_handler]
+
+    return QueryBuilder(
+        prop_enum_cls=MockProperties,
+        client_side_handlers=client_handlers,
         server_side_handler=mock_server_side_handler,
     )
-    return instance
 
 
-@pytest.fixture(name="run_parse_where_test_case")
-def run_parse_where_test_case_fixture(instance, mock_server_side_handler):
+@pytest.fixture(name="run_parse_where_test")
+def run_parse_where_test_fixture(
```

Comment:
Nit: this fixture has the name test too

IMO this should be something like `parse_where_runner` to make it clear it's a test helper or runner rather than a test that's run directly

Reply:
Also any reason this is a fixture, rather than just a free function?

Reply:
`run_parse_where_test_fixture` uses fixtures so I assumed it had to also be a fixture?
I'm leaving it like it is for now - but I'm renaming the fixture to `parse_where_runner`

Reply:
Ah genuine apologies I couldn't see the decorator for the fixture on my review :man_facepalming: I'll take this either way since it's my mistake